File: workInfo/plugin/MethodClassInfo.java
package com.example.demo.pojo.plugin;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * 方法类信息
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class MethodClassInfo {
    /**
     * 方法类id
     * 系统生成系统与用户均不可修改
     */
    private String id;

    /**
     * 类描述
     * 系统生成默认为空用户可修改
     */
    private String description;

    /**
     * 方法类版本id
     * 系统生成系统与用户均不可修改
     */
    private String versionId;

    /**
     * 类全限定名
     * 系统生成系统与用户均不可修改
     */
    private String className;

    /**
     * 简单类名
     * 系统生成系统与用户均不可修改
     */
    private String simpleClassName;

    /**
     * 包名
     * 系统生成系统与用户均不可修改
     */
    private String packageName;

    /**
     * 该类中的方法列表
     */
    private List<MethodInfo> methods;
}

File: src/models/MethodClassInfo.js
/**
 * 方法类信息模型
 */
import { MethodInfo } from './MethodInfo';

export class MethodClassInfo {
  /**
   * 方法类id
   * 系统生成系统与用户均不可修改
   */
  id = '';

  /**
   * 类描述
   * 系统生成默认为空用户可修改
   */
  description = '';

  /**
   * 方法类版本id
   * 系统生成系统与用户均不可修改
   */
  pluginVersionId = '';

  /**
   * 类全限定名
   * 系统生成系统与用户均不可修改
   */
  className = '';

  /**
   * 简单类名
   * 系统生成系统与用户均不可修改
   */
  simpleClassName = '';

  /**
   * 包名
   * 系统生成系统与用户均不可修改
   */
  packageName = '';

  /**
   * 该类中的方法列表
   */
  methods = [];

  constructor(data = {}) {
    Object.assign(this, data);
    if (data.methods) {
      this.methods = data.methods.map(method => new MethodInfo(method));
    }
  }
}